### ISSUE TITLE
Access token API provides error

### ISSUE THREAD

**Issue by sridhar391** [RestyaPlatform/board#4137]
## Context
As one of the user mailed us about Access token API provides error

**Comment by MaminHakim** [comment]
after Installing SSL on the server and add Nginx configuration. like this 
```
ssl_certificate    /etc/letsencrypt/live/mt/fullchain.pem;
 ssl_certificate_key    /etc/letsencrypt/live/mt/privkey.pem;
    ssl_session_cache shared:SSL:1m;
    ssl_session_timeout  15m;
    # Set system wide SSL settings
    ssl_prefer_server_ciphers on;
    ssl_protocols TLSv1.2 TLSv1.1 TLSv1;
    #Forward Secrecy
    ssl_ciphers "EECDH+ECDSA+AESGCM EECDH+aRSA+AESGCM EECDH+ECDSA+SHA384 EECDH+ECDSA+SHA256 EECDH+aRSA+SHA384 EECDH+aRSA+SHA256 EECDH+aRSA+RC4 EECDH EDH+aRSA RC4 !aNULL !eNUL$
```
after a couple of minutes, I got a 401 error 
![image](https://user-images.githubusercontent.com/13296665/94375044-962ad180-011d-11eb-833b-dc3cb955b957.png)

if I remove the cookies everything is working well but a little later again error's coming back!

without SSL, when the Nginx configuration was without doman name everything was good.

**Comment by sridhar391** [comment]
@MaminHakim Can you please attach the Nginx error log file named `error.log` file in the folder path `/var/log/nginx`

**Comment by MaminHakim** [comment]
@sridhar391  Here is the log
[error.log](https://github.com/RestyaPlatform/board/files/5296676/error.log)
.

**Comment by sridhar391** [comment]
@MaminHakim We have checked the `error.log` file and there is no related contents regarding the Restyaboard issue, can you please restart the Nginx server and try to access the Restyaboard URL and after trying to access the Restyaboard URL, please attach the Nginx error log file named `error.log` file in the folder path `/var/log/nginx`

**Comment by MaminHakim** [comment]
Restarted and after that, I got this log. 
[error (1).log](https://github.com/RestyaPlatform/board/files/5299719/error.1.log)

**Comment by sridhar391** [comment]
@MaminHakim Can you please uncomment the SSL configuration lines in the Restyaboard configuration file `restyaboard.conf` and restart the Nginx server and try to access the Restyaboard URL and after trying to access the Restyaboard URL, please attach the Nginx error log file named `error.log` file in the folder path `/var/log/nginx`

**Comment by MaminHakim** [comment]
@sridhar391 
this log is after installing fresh Restyaboard on the server. without any SSL just with an IP address. 
```
2020/09/30 23:15:09 [error] 496#496: *9 FastCGI sent in stderr: "PHP message: PHP Warning:  Illegal string offset 'difference' in /usr/share/nginx/html/restyaboard/server/php$
2020/09/30 23:16:06 [error] 496#496: *14 FastCGI sent in stderr: "PHP message: PHP Notice:  Trying to access array offset on value of type bool in /usr/share/nginx/html/resty$
2020/10/01 04:00:46 [error] 496#496: *163 open() "/usr/share/nginx/html/restyaboard/client/GponForm/diag_Form" failed (2: No such file or directory), client: 42.239.12.88, se$
2020/10/01 05:11:40 [error] 496#496: *165 open() "/usr/share/nginx/html/restyaboard/client/solr/admin/info/system" failed (2: No such file or directory), client: 176.113.115.$
2020/10/01 09:09:37 [emerg] 1190#1190: bind() to 0.0.0.0:80 failed (98: Address already in use)
2020/10/01 09:09:37 [emerg] 1190#1190: bind() to 0.0.0.0:80 failed (98: Address already in use)
2020/10/01 09:09:37 [emerg] 1190#1190: bind() to 0.0.0.0:80 failed (98: Address already in use)
2020/10/01 09:09:37 [emerg] 1190#1190: bind() to 0.0.0.0:80 failed (98: Address already in use)
2020/10/01 09:09:37 [emerg] 1190#1190: bind() to 0.0.0.0:80 failed (98: Address already in use)
2020/10/01 09:09:37 [emerg] 1190#1190: still could not bind()
2020/10/01 10:45:13 [error] 3166#3166: *1 FastCGI sent in stderr: "PHP message: PHP Warning:  Illegal string offset 'difference' in /usr/share/nginx/html/restyaboard/server/p$
2020/10/01 11:01:34 [error] 3166#3166: *174 open() "/usr/share/nginx/html/restyaboard/client/robots.txt" failed (2: No such file or directory), client: 62.113.214.39, server:$
2020/10/01 11:01:47 [error] 3166#3166: *174 open() "/usr/share/nginx/html/restyaboard/client/robots.txt" failed (2: No such file or directory), client: 62.113.214.39, server:$
```

**Comment by sridhar391** [comment]
@MaminHakim Can you please send us the `restyaboard_install.log` file, if you have installed the Restyaboard using the Restyaboard installation script file `restyaboard.sh` or can you please tell us the steps you have followed to install the Restyaboard

**Comment by MaminHakim** [comment]
@sridhar391 Here is the restyaboard install log 
[restyaboard_install.log](https://github.com/RestyaPlatform/board/files/5327483/restyaboard_install.log)

**Comment by sridhar391** [comment]
@MaminHakim Please execute the below commands in the terminal one by one and please tell us if the Restyaboard URL is loading the Login page in the browser

````
systemctl stop apache2
systemctl start nginx
````

**Comment by MaminHakim** [comment]
@sridhar391 I've done it before. restyaboard is working now. but clients faced another problem. after 7 days after installation. they faced with a white screen. I think It regards to the cookies or something else. because when they trying on another system everything is well. on incgnito mode is also good.

**Comment by MaminHakim** [comment]
this is a new error.log 
[Uploading error.log.txt…]()

**Comment by sridhar391** [comment]
@MaminHakim Please upload the `errror.log` file correctly

**Comment by MaminHakim** [comment]
@sridhar391 
[error (3).log](https://github.com/RestyaPlatform/board/files/5393378/error.3.log)

**Comment by rrjanbiah** [comment]
*Internal note*

Relevant log:
`2020/10/14 14:30:07 [error] 3166#3166: *19482 FastCGI sent in stderr: "PHP message: PHP Notice:  Trying to access array offset on value of type bool in /usr/share/nginx/html/restyaboard/server/php/bootstrap.php on line 59" while reading response header from upstream, client: 127.0.0.1, server: 127.0.0.1, request: "GET /api/v1/users/3/activities.json?type=all&last_activity_id=357&direction=ASC&token=3acb85fbf2c32de688cd9452f5416334cf9b2004&_=1602673206674 HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.4-fpm.sock:", host: "example.com", referrer: "https://example.com/"`

Problematic line is https://github.com/RestyaPlatform/board/blob/master/server/php/bootstrap.php#L59

Looks like the tokens are already getting cleared.

**Comment by sridhar391** [comment]
@MaminHakim Fixed the issue in the commit https://github.com/RestyaPlatform/board/pull/4207/commits/a979ec11a92c6c6f9f7c3cbf3ebe2fa5fb8865a4 and the fix will be available in the upcoming release version v0.7 (Scorpions)
